src/api/03-user-experience/08-picture-in-picture.md

```diff
@@ -0,0 +1,134 @@
+
+```html
+<video src="video.mp4"></video>
+<!-- Disable the button by default. -->
+<button disabled>Picture-in-Picture not available</button>
+
+<script>
+  const video = document.querySelector('video');
+  const button = document.querySelector('button');
+
+  const setupPip = () => {
+    // Only enable the button when PiP is enabled.
+    if (!document.pictureInPictureEnabled) return;
+
+    button.innerText = 'Toggle Picture-in-Picture';
+    button.disabled = false;
+
+    button.addEventListener('click', async () => {
+      // Entering and leaving PiP could take a little while, so we'll
+      // disable the button while we wait.
+      button.disabled = true;
+      try {
+        document.pictureInPictureElement === video
+          ? await document.exitPictureInPicture()
+          : await video.requestPictureInPicture();
+      } catch (e) {
+        // Show or log the error.
+      }
+      button.disabled = false;
+    });
+  }
+  setupPip();
+</script>
+```
+
+## Advanced usage
+
+There are several more things you might want to consider to make the Picture-in-Picture experience even better:
+
+1. Use the [Media Session API] action handlers to display more controls in the Picture-in-Picture window.
+2. Listen to `enterpictureinpicture` and `leavepictureinpicture` events and update your UI when Picture-in-Picture window is shown or closed.
+3. Display a [MediaStream] object such as the user's webcam feed, another application window or canvas contents in the Picture-in-Picture window.
+
+You can find more information and code snippets in a [Google Developers blog post].
+
+## See it in action
+
+When you play the video above, you can click the <span style="border-radius: 8px; background-color: var(--accent-background); width: 48px; height: 48px; display: inline-grid; place-items: center;" aria-label="Toggle picture in picture"><svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M20.25 15v6a1.5 1.5 0 0 1-1.5 1.5H3A1.5 1.5 0 0 1 1.5 21V8.25A1.5 1.5 0 0 1 3 6.75h3.75" stroke="var(--accent)" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"></path><path d="M9.75 3v7.5a1.5 1.5 0 0 0 1.5 1.5H21a1.5 1.5 0 0 0 1.5-1.5V3A1.5 1.5 0 0 0 21 1.5h-9.75A1.5 1.5 0 0 0 9.75 3Z" stroke="var(--accent)" stroke-width="1.5" stroke-miterlimit="10" stroke-linecap="square"></path><path d="M9 18.75V15H5.25M5.25 18.75 9 15" stroke="var(--accent)" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"></path></svg></span> button in the top right corner of the video to toggle Picture-in-Picture playback – if your browser supports the API.
+
+## What's next?
+
+Playing videos in a floating window is useful for a local viewing session. However users ofter prefer to watch content on other devices such as their TVs. In the [next article] we'll talk about media casting – and Google Cast in particular.
```

Comment:
Nit: `s/ofter/often`

Reply:
Updated in 281290836b44cdc445dea0ef7410d1082e9d8506.